internal/controllers/common_utils.go

```diff
@@ -40,32 +40,13 @@ func CommonLabelsAsLabelSelector(addon *addonsv1alpha1.Addon) labels.Selector {
 }
 
 // Tests if the controller reference on `wanted` matches the one on `current`
-func HasEqualControllerReference(current, wanted metav1.Object) bool {
-	currentOwnerRefs := current.GetOwnerReferences()
-
-	var currentControllerRef *metav1.OwnerReference
-	for _, ownerRef := range currentOwnerRefs {
-		or := ownerRef
-		if *or.Controller {
-			currentControllerRef = &or
-			break
-		}
-	}
-
-	if currentControllerRef == nil {
+func HasSameController(current, wanted metav1.Object) bool {
```

Comment:
```suggestion
func HasSameController(objA, objB metav1.Object) bool {
```

Doesn't matter what the parameter names are changed to, but the old ones imply that this is not an associative function.

Reply:
Really good point :ok_hand: